### ISSUE TITLE
Problem reading Matlab file

### ISSUE THREAD

**Issue by mcrot** [ContactEngineering/topobank#905]
We have the following which happened during upload of a file, probably coming from Matlab:
```
Internal Server Error: /manager/surface/1585/new-topography/

TypeError at /manager/surface/1585/new-topography/
'NoneType' object is not subscriptable

Request Method: POST
Request URL: https://contact.engineering/manager/surface/1585/new-topography/?redirect=/manager/surface/1585/&X-Progress-ID=e8eb89c2-4c84-4f92-b22b-758192888101
Django Version: 3.2.15
Python Executable: /opt/conda/bin/python
Python Version: 3.9.7
Python Path: ['/app', '/app', '/opt/conda/bin', '/opt/conda/lib/python39.zip', '/opt/conda/lib/python3.9', '/opt/conda/lib/python3.9/lib-dynload', '/opt/conda/lib/python3.9/site-packages', '/app/topobank']
Server time: Mon, 10 Oct 2022 23:04:38 +0200
Installed Applications:
['django.contrib.auth',
 'django.contrib.contenttypes',
 'django.contrib.sessions',
 'django.contrib.sites',
 'django.contrib.messages',
 'whitenoise.runserver_nostatic',
 'collectfast',
 'django.contrib.staticfiles',
 'django.contrib.admin',
 'django.contrib.postgres',
 'crispy_forms',
 'allauth',
 'allauth.account',
 'allauth.socialaccount',
 'allauth.socialaccount.providers.orcid',
 'rest_framework',
 'fontawesomefree',
 'formtools',
 'bokeh',
 'termsandconditions',
 'storages',
 'guardian',
 'bootstrap_datepicker_plus',
 'django_select2',
 'django_tables2',
 'progressbarupload',
 'celery_progress',
 'notifications',
 'django_filters',
 'tagulous',
 'trackstats',
 'fullurl',
 'topobank.users.apps.UsersAppConfig',
 'topobank.manager.apps.ManagerAppConfig',
 'topobank.analysis.apps.AnalysisAppConfig',
 'topobank.usage_stats.apps.UsageStatsAppConfig',
 'topobank.tabnav.apps.TabNavAppConfig',
 'topobank.publication.apps.PublicationAppConfig',
 'topobank.taskapp.celery.CeleryAppConfig',
 'gunicorn']
Installed Middleware:
['django.middleware.security.SecurityMiddleware',
 'whitenoise.middleware.WhiteNoiseMiddleware',
 'django.contrib.sessions.middleware.SessionMiddleware',
 'django.middleware.common.CommonMiddleware',
 'django.middleware.csrf.CsrfViewMiddleware',
 'django.contrib.auth.middleware.AuthenticationMiddleware',
 'termsandconditions.middleware.TermsAndConditionsRedirectMiddleware',
 'topobank.middleware.anonymous_user_middleware',
 'django.contrib.messages.middleware.MessageMiddleware',
 'django.middleware.clickjacking.XFrameOptionsMiddleware',
 'topobank.usage_stats.middleware.count_request_middleware']


Traceback (most recent call last):
  File "/opt/conda/lib/python3.9/site-packages/django/core/handlers/exception.py", line 47, in inner
    response = get_response(request)
  File "/opt/conda/lib/python3.9/site-packages/django/core/handlers/base.py", line 181, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/opt/conda/lib/python3.9/contextlib.py", line 79, in inner
    return func(*args, **kwds)
  File "/opt/conda/lib/python3.9/site-packages/django/contrib/auth/decorators.py", line 21, in _wrapped_view
    return view_func(request, *args, **kwargs)
  File "/opt/conda/lib/python3.9/site-packages/django/views/generic/base.py", line 70, in view
    return self.dispatch(request, *args, **kwargs)
  File "/opt/conda/lib/python3.9/site-packages/django/contrib/auth/mixins.py", line 128, in dispatch
    return super().dispatch(request, *args, **kwargs)
  File "/opt/conda/lib/python3.9/site-packages/formtools/wizard/views.py", line 244, in dispatch
    response = super().dispatch(request, *args, **kwargs)
  File "/opt/conda/lib/python3.9/site-packages/django/views/generic/base.py", line 98, in dispatch
    return handler(request, *args, **kwargs)
  File "/opt/conda/lib/python3.9/site-packages/formtools/wizard/views.py", line 291, in post
    form = self.get_form(data=self.request.POST, files=self.request.FILES)
  File "/opt/conda/lib/python3.9/site-packages/formtools/wizard/views.py", line 411, in get_form
    kwargs = self.get_form_kwargs(step)
  File "/app/topobank/manager/views.py", line 301, in get_form_kwargs
    channel_infos = step0_data['channel_infos']

Exception Type: TypeError at /manager/surface/1585/new-topography/
Exception Value: 'NoneType' object is not subscriptable
```
So the `channel_infos` are missing in the reader.

Here some more metadata:
```
topography_create_wizard-current_step = 'metadata'
metadata-data_source = '0'
metadata-name = 'Q51_100X_3x3.mat'
metadata-measurement_date = '2022-10-09'
metadata-description = '“Sample Q51” The surface-topography challenge. Raw measurement of the 3D topography of the surface using a KEYENCE VK-100 series microscope, at X100 magnification. (Truncated);'
metadata-tags = ''
save = 'Next'
redirect = '/manager/surface/1585/'
```
We should check our Matlab readers.

**Comment by pastewka** [comment]
I don't understand how this can happen, in particular since the data seems to be assigned here where it is explicitly converted to a list:

https://github.com/ContactEngineering/TopoBank/blob/ae68fd315235767ad7d65d8c6fecf0ae4fe4bf54/topobank/manager/forms.py#L185

**Comment by pastewka** [comment]
Sorry, `step0_data` itself is None.

**Comment by pastewka** [comment]
Resolved with #1012
